### PR TITLE
add conformance tests

### PR DESCRIPTION
**Pull request by noot**
## Changes

- add `verify_conformance_test` which verifies signatures generated by the go implementation
- port https://github.com/xx-labs/sleeve/blob/main/wots/key_test.go#L549 to `compute_public_key_conformance_test`
- fix `Key::from_seed`

## Tests

```
cargo test
```

## Issues

- n/a

### CODE REVIEW

File: src/params.rs [hunk 1]
```diff
@@ -273,7 +270,7 @@ impl<PRFH: Hasher + Clone, MSGH: Hasher + Clone> Params<PRFH, MSGH> {
 
             let mut hasher = PRFH::new();
             hasher.write(p_seed.to_vec());
-            hasher.write(vec![j as u8]);
+            hasher.write(vec![(j + 1) as u8]);
```
Comment (clearloop) [inline src/params.rs:273]: Is this `j + 1` always `< 255`?

File: src/params.rs [hunk 2]
```diff
@@ -358,7 +355,7 @@ fn parity(value: &[u8]) -> bool {
         v ^= v >> 4;
         v ^= v >> 2;
         v ^= v >> 1;
-        count += (v & 1) as i8
+        count += (v & 1) as i64
```
Comment (clearloop) [inline src/params.rs:358]: same here

**Review by clearloop — APPROVED** [review]
All good! but the usage of `as` in rust may have potential risks, for example `256_i16 as u8` will be `255_u8`, `u8 as i16` is safe, but `i16 as u8` is not

File: src/params.rs (hunk 1 above)
Comment (noot) [reply src/params.rs:273]: yes, the highest it can go is `(W-1)` which is 255

File: src/params.rs (hunk 2 above)
Comment (noot) [reply src/params.rs:358]: should be ok since it's a byte being cast to i64

**Pull request merged**